src/Stratis.Bitcoin/Utilities/BinarySearch.cs

```diff
@@ -17,8 +17,10 @@ private static T BinaryFindFirst<T>(T[] array, Func<T, bool?> func, int first, i
 
             // Otherwise split the array in two and search each half.
             int pivot = length / 2;
-            return BinaryFindFirst(array, func, first, pivot)
-                ?? BinaryFindFirst(array, func, first + pivot, length - pivot);
+            var result = BinaryFindFirst(array, func, first, pivot);
```

Comment:
@quantumagi the solution didnt build this my side. It was due to a coalescing feature only available in C#8